pkg/convert/tf.go

```diff
@@ -787,6 +795,58 @@ func (s *convertState) sourceCode(rng hcl.Range) string {
 	return strings.Replace(buffer.String(), "\r\n", "\n", -1)
 }
 
+func (s *convertState) renamePclOverlap(kind string, hclType *string, name string, hclRange *hcl.Range) (string, string) {
+	newName := name
+	newType := new(string)
+	if hclType != nil {
+		*newType = *hclType
+	} else {
+		*newType = ""
+	}
+
+	if hclType != nil && isPclKeyword(*hclType) {
+		*newType = fmt.Sprintf("%s_%sType_pcl", *hclType, kind)
+		s.typeRenames.nameToRename[name] = *newType
+
+		s.appendDiagnostic(&hcl.Diagnostic{
+			Subject:  hclRange,
+			Severity: hcl.DiagWarning,
+			Summary:  kind + " type renamed to prevent keyword overlap",
+			Detail:   fmt.Sprintf("Renaming %s type %s to %s to prevent overlap", kind, *hclType, *newType),
+		})
+	}
+
+	if isPclKeyword(name) {
+		newName = fmt.Sprintf("%s_%s_pcl", name, kind)
```

Comment:
PCL isn't really something we've publicly documented. It's really more of an internal implementation detail. I'm wondering if we should use something other than "pcl" for the suffix? Would a single underscore work or does it need to be alpha numeric?

Reply:
I'll give it a shot